### PR TITLE
SQL-2244: Add ARM support for linux in run_adf.sh and evergreen

### PR DESCRIPTION
**Pull request by bucaojit**
Made changes to support the Linux ARM variants in `run_adf.sh`.  
~I added the ubuntu variant to evergreen, the ADF team runs our tests in a docker container with an ubuntu base.~  Adding `amazon2-arm64-graviton3` as requested.
Passing integration tests is [here](https://spruce.mongodb.com/task/mongo_jdbc_driver_ubuntu2204_arm64_jdk_8_test_integration_patch_1a111b623a61a45c3b25e25f86e444d3369bd070_669ee3f103474c00077aa9da_24_07_22_22_57_54?execution=0&sortBy=STATUS&sortDir=ASC).

Once this is reviewed, I'll add changes to the other repos that have run_adf: mongo-powerbi-connector and mongo-odbc-driver

### CODE REVIEW

File: .evg.yml [hunk 1]
```diff
@@ -47,6 +47,16 @@ buildvariants:
       - name: "test-unit"
       - name: "test-integration"
 
+  - name: ubuntu2204-arm64-jdk-8
```
Comment (Utagai) [inline .evg.yml:50]: The distro we'd be trying this on closely matches what we actually would use in production, `amazon2-arm64-graviton3`. What do you think about swapping to this?

**Review by Utagai — COMMENTED** [review]
Just wanted to drop a thanks for your quick work on this! I appreciate it a lot! I dropped by a comment on the chosen distro that I noticed.

Also, while I can't do it this week I think (on rotation) but I might be able to test these changes as well on our task if I point it to your feature branch? Let me know...

File: .evg.yml (hunk 1 above)
Comment (terakilobyte) [inline .evg.yml:50]: Should this be amazon linux?

**Review by terakilobyte — COMMENTED** [review]
A question about the distro

File: .evg.yml (hunk 1 above)
Comment (bucaojit) [reply .evg.yml:50]: Sure, I started up an `amazon2-arm64-graviton3` host to test it out.  I'll likely add another check for this distro with different links.

Comment (bucaojit) [reply .evg.yml:50]: I converted the PR to draft to work on adding amazon linux to address the latest comments.

File: .evg.yml [hunk 2]
```diff
@@ -319,7 +329,7 @@ functions:
       working_dir: mongo-jdbc-driver
       script: |
           ${PREPARE_SHELL}
-          ./gradlew clean -x test -x integrationTest spotlessApply build --rerun-tasks
+          ./gradlew -Dorg.gradle.java.home=${JAVA_HOME} clean -x test -x integrationTest spotlessApply build --rerun-tasks
```
Comment (bucaojit) [inline .evg.yml:332]: The `amazon2-arm64-graviton3` host wasn't picking up the `$JAVA_HOME` variable in gradle automatically.  Setting it explicitly here works.

File: .evg.yml [hunk 3]
```diff
@@ -405,16 +415,19 @@ functions:
...
-          case '${_platform}' in
+          case ${_platform} in
           ubuntu2204-64-jdk-8)
                JAVA_HOME=/opt/java/jdk8
           ;;
           ubuntu2204-64-jdk-11)
                JAVA_HOME=/opt/java/jdk11
           ;;
+          amazon2-arm64-jdk-11)
+               JAVA_HOME=/usr/lib/jvm/java-11
```
Comment (bucaojit) [inline .evg.yml:426]: There isn't a jdk8 installed on the host, only this one.  If we want 8 I can create a devprod ticket.

File: resources/run_adf.sh [hunk 4]
```diff
@@ -82,10 +83,15 @@ MONGOSH_DOWNLOAD_BASE=https://downloads.mongodb.com/compass
...
 # RedHat 7
 MONGO_DOWNLOAD_REDHAT=mongodb-linux-x86_64-rhel70-7.0.0-rc6.tgz
+# RedHat 9 ARM
+MONGO_DOWNLOAD_REDHAT_ARM=mongodb-linux-aarch64-rhel90-7.0.0.tgz
 # Shared Linux mongosh
 MONGOSH_DOWNLOAD_LINUX_FILE=mongosh-1.8.0-linux-x64.tgz
+MONGOSH_DOWNLOAD_LINUX_ARM_FILE=mongosh-1.8.0-linux-arm64.tgz
 
 ## macOS
 MONGO_DOWNLOAD_MAC=mongodb-macos-x86_64-7.0.0-rc6.tgz
```
[the remark this answers is not in the document]
Comment (bucaojit) [reply resources/run_adf.sh]: I made the change and ubuntu passes, I think it should be good.

**Comment by bucaojit** [comment]
Not sure why the Amazon Linux variant hangs on the integration test.  It runs fine when I run it on an Evergreen host, but I also wasn't hitting the `JAVA_HOME` env variable issue there either.  I'll have to look into that a bit more.

File: .evg.yml (hunk 3 above)
Comment (nbagnard) [reply .evg.yml:426]: I think we should be able to build with java 11, I'm pretty sure that I build with 13.

File: resources/run_adf.sh [hunk 5]
```diff
@@ -155,7 +161,11 @@ check_mongod() {
 # Check if jq exists.  If not, download and set path
 get_jq() {
   if [ $OS = "Linux" ]; then
-    curl -L -o $JQ https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
+    if [ "$MACHINE_ARCH" = "aarch64" ]; then
+      curl -L -o $JQ https://mongosql-noexpire.s3.us-east-2.amazonaws.com/run_adf/jq-linx-arm
```
[the remark this answers is not in the document]
Comment (bucaojit) [reply resources/run_adf.sh:168]: Whoops.  Okay I'll leave it for now.

**Comment by bucaojit** [comment]
I tried a number of things to try and figure out why the integration test is hanging.  It works if I run it manually and it works on ubuntu, it might be an Evergreen problem.  I asked on the devprod-evergreen channel for some pointers.

**Review by terakilobyte — APPROVED** [review]
LGTM

**Comment by bucaojit** [comment]
I created SQL-2248 to address the integration test hang as I'll be out and I'd like to have this fix merged for May to test out.

**Pull request merged**